### PR TITLE
Clarify "Push to stage" dialog env var statement

### LINKED ISSUE #4603: Confusing statement about environment variables in the pipeline "push to stage" dialog
### Current Behavior

In reviewing an article around pipelines, the authors [write up regarding how Env Vars are transmitted](https://github.com/FlowFuse/website/pull/2603/files#diff-b85a0880c14c02bdda69e8e7da45a0ef81b637375fe0f09c8eafb991e20957bcR82) did not match my expectation of how things operate. I double checked this and found the write up to be incorrect but it was a reasonable interpretation of the dialog we show to users.

image

Last sentence states `It will also transfer the keys, but not the values, of any newly created Environment Variables not currently in the target instance.`

I checked with the team on slack and unsurprisingly got misunderstanding across the board!

### Expected Behavior

clarify the statement.

Something like  "The pipeline operation will copy environment variables but it will not overwrite any that exist in the target."

something that describes the visualisation below:

image

### Steps To Reproduce

operate a pipeline stage

### Environment

- FlowFuse version: 2.9 - prod/live/staging
- Node.js version:
- npm version:
- Platform/OS:
- Browser:

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

### PR DESCRIPTION
**Pull request by Steve-Mcl**
closes #4603

## Description

Clarify "Push to stage" dialog env var statement

Before: 

image

After:

image

## Related Issue(s)

#4603 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Review by cstns — APPROVED** [review]

**Comment by cstns** [comment]
Feel free to merge after tests pass

**Comment by Steve-Mcl** [comment]
Thanks @cstns appreciate quick feedback.

Hoping to the OK from @joepavitt and perhaps @sumitshinde-84 too since it arose from confusion arose from the new article here: https://github.com/FlowFuse/website/pull/2603#discussion_r1784057978

**Review by sumitshinde-84 — APPROVED** [review]

**Comment by joepavitt** [comment]
First one states that `keys` will transfer, but the values will not. Now, the new one states that `keys` will not be overwritten, but that implies the values will?

**Comment by Steve-Mcl** [comment]
> Now, the new one states that `keys` will not be overwritten, but that implies the values will?

Thats not what i (and the other guys who discussed this on Slack) infer from it ;) but since you did, perhaps a simplification would be better?

> Environment variables in the target instance that already have a value will not be overwritten

Clean, to the point. 


<br>

More explicit?:
> Environment variables will be copied to the target instance but any that already have a value will not be overwritten.

<br>

Super precise?:

> Environment Variables will be copied from the source to the target instance. Any Environment Variables that the target already has will not be overwritten if they contain a value however empty Environment Variables will be updated.


Thoughts?

**Comment by joepavitt** [comment]
> Environment variables in the target instance that already have a value will not be overwritten

This please

File: frontend/src/components/pipelines/DeployStageDialog.vue [hunk 1]
```diff
@@ -34,8 +34,7 @@
                 </p>
             </template>
             <p class="my-4">
-                It will also transfer the keys, but not the values, of any newly created Environment
-                Variables not currently in the target {{ targetTypeName }}.
+                NOTE: Environment Variable keys that already exist in the target {{ targetTypeName }} will not be overwritten.
```
Comment (Steve-Mcl) [inline frontend/src/components/pipelines/DeployStageDialog.vue:37]:
```suggestion
                NOTE: Environment variables in the target {{ targetTypeName }} that already have a value will not be overwritten.
```

**Comment by Steve-Mcl** [comment]
> > Environment variables in the target instance that already have a value will not be overwritten
> 
> This please

Done diddly done.

 [a0e93c4](https://github.com/FlowFuse/flowfuse/pull/4604/commits/a0e93c4eb4abf9342411be111c1e088f5adb6476)

**Review by joepavitt — APPROVED** [review]

**Pull request merged**